src/Route.php

```diff
@@ -148,9 +148,7 @@ public function getAllowedMethods() : ?array
     public function allowsMethod(string $method) : bool
     {
         $method = strtoupper($method);
-        if (RequestMethod::METHOD_HEAD === $method
-            || RequestMethod::METHOD_OPTIONS === $method
-            || $this->methods === self::HTTP_METHOD_ANY
+        if ($this->methods === self::HTTP_METHOD_ANY
```

Comment:
When we backport this to the v2 release, we'll need to note the change in behavior, as it's technically a BC break, even if it's in the service of a bugfix.